js/base-tool-table.js

```diff
@@ -0,0 +1,393 @@
+				name: "",
+			};
+			
+			// Set the name
+			for (let col = 0; col < labels.length; col++) {
+				// Add a seperator for cases of multiple columns
+				if (out.name.length > 0) {
+					out.name += " | "
+				}
+				// Add each column to out.name
+				if (col !== dplace) {
+					// Get rid of ugly notation
+					const clean = Renderer.stripTags(i[col]);
+					out.name += clean;
+				}
+			}
+			
+			// Set the weight
+			if (dplace !== -1) {
+				const weight = i[dplace];
+				const dash = weight.indexOf("\u2013"); // Note: – is different from -
+				
+				// If the weight is a range
+				if (~dash) {
+					// Get the two numbers in the range, subtract them, add 1
+					const low = parseInt(weight.substring(0, dash));
+					let high = parseInt(weight.substring(dash + 1));
+					if (high === 0) high = 100;
+					out.weight = high - low + 1;
+				} else { // If the weight is a single value
+					out.weight = 1;
+				}
+			} else { // If the weight is unlisted
+				out.weight = 1;
+			}
+			
+			if (i.avatar) out.avatar = i.avatar;
+			return out;
+		}));
+		r20t.tableitems.forEach(it => it.save());
+	}
+	
+	// A function for generating and saving tables from JSONs of the Roll20 format
+	function createR20Format(t) {
+		const r20t = d20.Campaign.rollabletables.create({
+			name: t.name.replace(/\s+/g, "-"),
+			showplayers: t.isShown,
+			id: d20plus.ut.generateRowId(),
+		});
+		
+		r20t.tableitems.reset(t.items.map(i => {
+			const out = {
+				id: d20plus.ut.generateRowId(),
+				name: i.row,
+			};
+			if (i.weight !== undefined) out.weight = i.weight;
+			if (i.avatar) out.avatar = i.avatar;
+			return out;
+		}));
+		r20t.tableitems.forEach(it => it.save());
+	}
+	
+	// Function to create from create your own format
+	// Note that due to the simplicity of the format, hiding from players, weight, and avatars can't be configured
+	// Returns true on success, false on failure
+	function createCreateFormat(t) {
+		// Split the input into lines
+		const lines = t.split("\n");
+		
+		// Ensure there's a table name and at least one element
+		if (lines.length < 2) {
+			return [false, "Must contain table name and at least one element"];
+		}
+		
+		// Create table object, using lines[0] as the name
+		const r20t = d20.Campaign.rollabletables.create({
+			name: lines[0].replace(/\s+/g, "-"),
+			showplayers: true,
+			id: d20plus.ut.generateRowId(),
+		});
+		
+		// Create a row from each line
+		r20t.tableitems.reset(lines.slice(1).map(i => {
+			const out = {
+				id: d20plus.ut.generateRowId(),
+				name: i,
+			};
+			return out;
+		}));
+		
+		r20t.tableitems.forEach(it => it.save());
+
+		return [true, null];
+	}
+	
+	// Simplified version of getFromPaste which checks if the paste is valid
+	function validatePaste(paste) {
+		let valid = true;
+		let error = null;
+		let tbl = false;
+		
+		paste.split("\n").forEach(line => parseLine(line.trim()));
+		parseLine(""); // ensure trailing newline
+		return [valid, error];
```

Comment:
as above, prefer returning objects (or in this case, probably just return the error message on its own)